lib/api/dsl/methods.js

```diff
@@ -670,8 +670,53 @@ module.exports = function Methods(filtersObject) {
   /**
    * Return true only if the value in field pass the regexp test
    */
-  this.regexp = function () {
-    return q.reject(new KuzzleError('regexp is not implemented yet.'));
+  this.regexp = function (roomId, index, collection, filter, not) {
+    var
+      k,
+      encodedFunctionName,
+      field,
+      value,
+      result,
+      regexp,
+      flags;
+
+    if (typeof filter !== 'object') {
+      return q.reject(new BadRequestError('Regexp argument must be an object'));
+    }
+
+    for (k in filter) {
+      if (filter.hasOwnProperty(k)) {
+        field = k;
+        value = filter[k];
```

Comment:
If I understand correctly, this means that if there are multiple regular expressions defined, we silently take and process the last one.
Shouldn't it be safer to detect multiple regexp defined and throw a `BadRequestError`?